### PR TITLE
refactor(app): Factors out orca task polling to its own package.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

File: util/task_watcher.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	}
+	return nil
+}
+
+func taskCompleted(task map[string]interface{}) bool {
+	taskStatus, exists := task["status"]
+	if !exists {
+		return false
+	}
+gg
```
Comment (ezimanyi) [inline util/task_watcher.go]: Unintentional?

File: cmd/orca-tasks/task_watcher.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	id := idFromTaskRef(taskRef)
+	task, resp, err := gateClient.TaskControllerApi.GetTaskUsingGET1(gateClient.Context, id)
+
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return fmt.Errorf("Encountered an error saving application, status code: %d\n", resp.StatusCode)
+	}
+	if !taskSucceeded(task) {
```
Comment (ezimanyi) [inline cmd/orca-tasks/task_watcher.go]: It looks like if the task is still running, we won't even get to the retry logic...should this be removed entirely and instead fallback to the `taskSucceeded` check after we know the task is complete?

File: cmd/orca-tasks/task_watcher.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		return fmt.Errorf("Encountered an error saving application, task output was: %v\n", task)
+	}
+
+	attempts := 0
+	for (task == nil || !taskCompleted(task)) && attempts < maxAttempts {
+		id := idFromTaskRef(taskRef)
+		task, resp, err = gateClient.TaskControllerApi.GetTaskUsingGET1(gateClient.Context, id)
```
Comment (ezimanyi) [inline cmd/orca-tasks/task_watcher.go]: Is it deliberate that we don't check `err` or `respStatusCode` here?  (Maybe to just try again if there's a communication error?)

File: cmd/orca-tasks/task_watcher.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+		attempts += 1
+		time.Sleep(time.Duration(attempts*attempts) * time.Second)
```
Comment (ezimanyi) [inline cmd/orca-tasks/task_watcher.go]: This is minor, but we'll always sleep for some amount of time *after* the last request, so we're artificially making things take longer than they need to.  We probably want this at the beginning of the loop (which also will add a delay between the poll at line 28 and the first poll in the loop.)

Comment (ezimanyi) [reply cmd/orca-tasks/task_watcher.go]: Also, is this deliberately quadratic backoff?  I'm less concerned about the particular backoff algorithm, but the max wait time here is < 1 min for 5 attempts...is that enough?

File: util/task_watcher.go (hunk 1 above)
Comment (jtk54) [reply util/task_watcher.go]: Yep.

File: cmd/orca-tasks/task_watcher.go (hunk 2 above)
Comment (jtk54) [reply cmd/orca-tasks/task_watcher.go]: Agreed, I just removed all checks here and delegate to the normal while `for` loop.

File: cmd/orca-tasks/task_watcher.go (hunk 3 above)
Comment (jtk54) [reply cmd/orca-tasks/task_watcher.go]: `task == nil` handles the case where there is no retrieved task for any reason (network issues etc) which is why we're ignoring `resp` and `err` here.

File: cmd/orca-tasks/task_watcher.go (hunk 4 above)
Comment (jtk54) [reply cmd/orca-tasks/task_watcher.go]: I think the number of attempts is something we'll have to tune. This is just a heuristic that's been sufficient for now.

File: cmd/orca-tasks/task_watcher.go (hunk 3 above)
Comment (ezimanyi) [reply cmd/orca-tasks/task_watcher.go]: Makes sense :+1:

**Pull request merged**